coordinator/src/main/java/com/tencent/rss/coordinator/ClientConfManager.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.coordinator;
+
+import java.io.Closeable;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClientConfManager implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(ClientConfManager.class);
+
+  private final AtomicReference<Map<String, String>> clientConf = new AtomicReference<>();
+  private final AtomicLong lastCandidatesUpdateMS = new AtomicLong(0L);
+  private Path path;
+  private ScheduledExecutorService updateClientConfSES = null;
+  private FileSystem fileSystem;
+  private static final String WHITESPACE_REGEX = "\\s+";
+
+  public ClientConfManager(CoordinatorConf conf, Configuration hadoopConf) throws Exception {
+    if (conf.getBoolean(CoordinatorConf.COORDINATOR_DYNAMIC_CLIENT_CONF_ENABLED)) {
+      init(conf, hadoopConf);
+    }
+  }
+
+  private void init(CoordinatorConf conf, Configuration hadoopConf) throws Exception {
+    String pathStr = conf.get(CoordinatorConf.COORDINATOR_DYNAMIC_CLIENT_CONF_PATH);
+    this.path = new Path(pathStr);
+
+    this.fileSystem = CoordinatorUtils.getFileSystemForPath(path, hadoopConf);
```

Comment:
Would we put the conf in hdfs? If hdfs is broken,  we can't get the conf?

Reply:
This is a good point, for now we only check the file existence at the startup, I will add conf loading check since the conf file normally contains critical configurations.